### PR TITLE
compaction_manager: Don't let ENOSPC throw out of ::stop() method

### PR DESCRIPTION
**Pull request by xemul**
The seastar defer_stop() helper is cool, but it forwards any exception from the .stop() towards the caller. In case the caller is main() the exception causes Scylla to abort(). This fires, for example, in compaction_manager::stop() when it steps on ENOSPC

Signed-off-by: Pavel Emelyanov <xemul@scylladb.com>

### CODE REVIEW

**Review by bhalevy — APPROVED** [review]

File: main.cc [hunk 1]
```diff
@@ -1007,7 +1007,9 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 };
             });
             cm.start(std::move(get_cm_cfg), std::ref(stop_signal.as_sharded_abort_source())).get();
-            auto stop_cm = deferred_stop(cm);
+            auto stop_cm = defer_verbose_shutdown("compaction manager", [&cm] {
+                cm.stop().get();
```
Comment (avikivity) [inline main.cc:1011]: I think it's better to have compaction_manager::stop swallow any bad exception and not leak them to main.

**Comment by xemul** [comment]
:pause_button:

**Comment by xemul** [comment]
v2:
- swallow ENOSPC in compaction_manager::stop(), leave the main's defer_stop() call intact

:arrow_forward:

**Review by raphaelsc — APPROVED** [review]

**Review by bhalevy — APPROVED** [review]

**Comment by denesb** [comment]
Cover letter needs an update.

**Comment by denesb** [comment]
Actually, with the updated title it is fine.

**Comment by denesb** [comment]
@xemul in the future please set the target branch to master, so the PR is only closed after promotion.

**Pull request merged**